src/query/external/useSEContracStorage.ts

```diff
@@ -0,0 +1,79 @@
+import { useQuery } from "@tanstack/react-query";
+import { STELLAR_EXPERT_API } from "@/constants/settings";
+import { getStellarExpertNetwork } from "@/helpers/getStellarExpertNetwork";
+import { ContractStorageResponseItem, NetworkType } from "@/types/types";
+
+/**
+ * StellarExpert API to get smart contract’s storage data
+ */
+export const useSEContractStorage = ({
+  isActive,
+  networkId,
+  contractId,
+  totalEntriesCount,
+}: {
+  isActive: boolean;
+  networkId: NetworkType;
+  contractId: string;
+  totalEntriesCount: number | undefined;
+}) => {
+  const query = useQuery<ContractStorageResponseItem[] | null>({
+    queryKey: ["useSEContractStorage", networkId, contractId],
+    queryFn: async () => {
+      // No entries
+      if (!totalEntriesCount) {
+        return null;
+      }
+
+      const network = getStellarExpertNetwork(networkId);
+
+      if (!network) {
+        return null;
+      }
+
+      let allRecords: ContractStorageResponseItem[] = [];
+
+      const fetchData = async (cursor?: string) => {
+        const searchParams = new URLSearchParams();
+
+        searchParams.append("order", "asc");
+        searchParams.append("limit", "200");
+
+        if (cursor) {
+          searchParams.append("cursor", cursor);
+        }
+
+        const response = await fetch(
+          `${STELLAR_EXPERT_API}/${network}/contract-data/${contractId}?${searchParams.toString()}`,
+        );
+
+        const responseJson = await response.json();
+
+        if (responseJson.error) {
+          throw responseJson.error;
+        }
+
+        allRecords = [
+          ...allRecords,
+          ...(responseJson?._embedded?.records || []),
+        ];
+      };
+
+      try {
+        while (allRecords.length < totalEntriesCount) {
```

Comment:
a blend smart contract `CDVQVKOY2YSXS2IC7KN6MNASSHPAO7UN2UR2ON4OI2SKMFJNVAMDX6DP` has `5987 entries`. Upon initial fetch, there is some loading time. What do you think of batch fetch data call instead fetch data all entries?

<img width="758" alt="blend" src="https://github.com/user-attachments/assets/7bf4285b-937d-4444-a495-98204fb568ab" />

Reply:
Unfortunately, I don't think we can batch the calls right now. :disappointed: If we're doing sorting, filters, and search on the client, we need to have all the data to make it work as expected. If this becomes an issue, we might look into some clever things to do with batching, but that will be quite complex.